src/compiler/parser.ts

```diff
@@ -1891,7 +1891,15 @@ namespace Parser {
                     }
 
                     if (pos >= 0) {
-                        const nonAwaitStatement = sourceFile.statements[pos];
+                        let nonAwaitStatement = sourceFile.statements[pos];
+                        // skip all statements in range already completely consumed by the reparsed statement
+                        for (let i = pos + 1; i < sourceFile.statements.length; i++) {
+                            if (nonAwaitStatement.end >= statement.end) {
+                                break;
+                            }

```

Comment:
```suggestion
                        for (let i = pos + 1; i < sourceFile.statements.length && sourceFile.statements[i].end < statement.end; i++) {
```

but it would actually be better to have a loop that increments pos as far as it needs to go and ends with `const nonAwaitStatement = sourceFile.statements[pos]`.